### ISSUE TITLE
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by brianjmurrell** [microsoft/vscode-docker#3268]
Issue Type: <b>Bug</b>

I keep getting a constant stream of errors from the Dockerfile Language Server:

```
  Code: -32603 
[Error - 1:23:45 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:23:46 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:05 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:06 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:09 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:10 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:11 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:12 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:16 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:17 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:19 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:19 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:24:24 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:45:50 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:47:18 p.m.] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
```

Extension version: 1.17.0
VS Code version: Code 1.61.1 (c13f1abb110fc756f9b3a6f16670df9cd9d4cf63, 2021-10-14T01:15:35.620Z)
OS version: Linux x64 5.14.9-100.fc33.x86_64
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz (12 x 3514)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 4, 4|
|Memory (System)|62.57GB (1.06GB free)|
|Process Argv|--unity-launch --crash-reporter-id d33a766b-28e9-45e5-942d-3a9085ddc0c0|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
vsccppwtct:30382698
pythonvssor306:30344512
bridge0708:30335490
pygetstartedt2:30371810
dockerwalkthru:30377721
bridge0723:30353136
pythonrunftest32:30373476
pythonf5test824:30373475
javagetstartedt:30364666
pythonvspyt187:30373474
vsqsis300:30381676
vsaa593:30376534
vssld246:30379911

```

</details>

**Comment by rcjsuen** [comment]
> [Dockerfile](https://github.com/daos-stack/daos/blob/98003f0c9851a7fac0423a21e97b1ec70a3bd9aa/utils/docker/Dockerfile.centos.8) is doing it here for me.

Thank you for that. I was able to reproduce the issue. It looks different than the one I just found.

> So is the implication that there is something erroneous in my [Dockerfile](https://github.com/daos-stack/daos/blob/98003f0c9851a7fac0423a21e97b1ec70a3bd9aa/utils/docker/Dockerfile.centos.8) (that I could fix) that is causing this?

Not really. If your Dockerfile builds it should be valid. If it doesn't build...the editor still shouldn't go crazy and spam you with error messages. :)

As I know how annoying the error messages can be, you can try disabling semantic highlighting in your `settings.json` file to stop it from spamming you.
```JSON
"[dockerfile]": {
    "editor.semanticHighlighting.enabled": false
}
```

**Comment by brianjmurrell** [comment]
> Thank you for that. I was able to reproduce the issue.

Excellent.  Please ensure that any PR to fix references this issue so that I know when I can re-enable `editor.semanticHighlighting.enabled`.

In fact it would be nice if the VS Code (or plugin) update that fixes this checks if the user has that setting disabled explicitly and asks them if they want to re-enable given that the issue is fixed and the OUTPUT log spamming should stop.

> ```json
> "[dockerfile]": {
>     "editor.semanticHighlighting.enabled": false
> }
> ```

Thanks for that.  Per the above, it would be nice to know if/when I can re-enable that without having to follow PR landings and subsequent releases, etc.

**Comment by bwateratmsft** [comment]
This fix is now released in Docker extension version 1.18.0.
